### DISCUSSION TITLE
Authentication fails with password hashing generated using bcrypt package in Node

### DISCUSSION THREAD

**Discussion by sakthivelpvs** [vernemq/vernemq#2091] [Q&A] [answered]
I have used bcrypt (https://www.npmjs.com/package/bcrypt) package in my nodejs code to generate hashing. The generated hash is stored in MongoDB `vmq_acl_auth` table and when I connect `MQTT client` it fails to connect broker. If I generate hash using https://bcrypt-generator.com/ and update the generate hashing in MongoDB, now the MQTT client gets connected.

Is there any supported node package for `VerneMQ` password hashing or any functionality available with `VerneMQ & MongoDB` to generate hashing automatically.

Example source that used for hash generating

```
const bcrypt = require('bcrypt');
token = "AddsdWW2333dsd"
let hash = bcrypt.hash(token, 6, async (err, hash) => {
  console.log(hash);
})
```

**Comment by mths1** [comment, marked as answer]
Please check that the library generates a 2a hash, as per documentation

"Note that currently bcrypt version `2a` (prefix `$2a$`) is supported."

**Reply by sakthivelpvs** [reply]
Thanks @mths1 . It works with following 

```
bcrypt.hash(token, bcrypt.genSaltSync(6, 'a')).then(hash => {
   console.log(hash);
});
```
